flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -2025,6 +2047,123 @@ private void freeNoLongerUsedSlots(AllocatedSlotReport allocatedSlotReport) {
         }
     }
 
+    private void tryPersistAllocationSnapshot(
+            SlotID slotId,
+            JobID jobId,
+            String jobTargetAddress,
+            AllocationID allocationId,
+            ResourceProfile resourceProfile) {
+        if (!allocationsDirectory.exists() && !allocationsDirectory.mkdirs()) {
+            log.debug(
+                    "Allocations directory {} doesn't exist and cannot be created.",
+                    allocationsDirectory.toPath());
+            return;
+        }
```

Comment:
Can we create the directory when we start the `TaskExecutor`? Then we don't have these checks here.

Reply:
I think it's safer to check it every time and eventually create it only if we need it, wdyt?

Reply:
If we assume that the directory can disappear in between, then yes. But also in this case, this solution is not bullet-proof.

Reply:
> If we assume that the directory can disappear in between, then yes

That's exactly what I was thinking of, because this is a best effort recovery system, I can imagine how the user might delete these files by mistake or to free up some space. The solution here is not bulletproof, and in fact if the user deletes the directory between this line and the lines below writing the file, the writing will fail. But it doesn't really matter, because again that's a best effort solution